WordPress/gutenberg
# Navigation Editor: Fix saving locations using the "Manage Locations" popup

## Description

This PR aims to give users the ability to assign all available menus to an arbitrary location inside the `Manage Locations` popup.
Currently it's only possible to assign a location to a menu that is being edited.
Fixes #33815.
This PR doesn't fix the issue with the main "Save" button because we have [another issue](https://github.com/WordPress/gutenberg/issues/31813) for that.

## How has this been tested?
1. This has been tested using the `Twenty Twenty-One` WordPress theme.
2. Go to the experimental Navigation screen.
3. Create two menus.
4. Switch to the first menu using the `Switch Menu` link at the top of the page.
5. Open the `Manage Locations` popup.
![Screenshot 2021-09-10 at 10 47 40](https://user-images.githubusercontent.com/43744263/132827409-84e29b78-58cc-48d4-84d0-0c0cb3359dfb.png)
6. Assign the first location to the first menu.
7. Assign the second location to the second menu.
8. Click the `Update` button.
![Screenshot 2021-09-10 at 10 48 21](https://user-images.githubusercontent.com/43744263/132828156-530f7d86-fe8d-48a5-8cc3-9ae48abcbcc8.png)

9. You should see `Menu locations have been updated` message at the bottom of the page.
![Screenshot 2021-09-10 at 10 48 18](https://user-images.githubusercontent.com/43744263/132827362-9e7281e9-a662-4b43-b058-8b6362daeda4.png)

10. Refresh the page.
11. Make sure that the first menu is assigned to the first location.
12. Switch to the second menu using the `Switch Menu` link at the top of the page.
13. Make sure that the second menu is assigned to the second location. 

## Types of changes
Bug fix ( but this also can be considered as a new feature).

## Checklist:
